libnetwork/iptables/firewalld.go

```diff
@@ -159,13 +197,44 @@ func OnReloaded(callback func()) {
 }
 
 // Call some remote method to see whether the service is actually running.
-func checkRunning() bool {
+func checkRunning() (bool, error) {
+
+	var currentState dbus.Variant
+	var wg sync.WaitGroup
+	var cancel context.CancelFunc
+	var ctx context.Context
+	var dbusTimeout = true
+	var err error
+
 	if connection == nil {
-		return false
+		return false, nil
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), DbusConnectionTimeout)
+	defer cancel()
+
+	//This is more complicated but without this, if polkit has an error dbus calls can hang for extended periods of time
+	wg.Add(1)
+
+	go func(ctx context.Context) {
+		defer wg.Done()
+		currentState, err = connection.sysObj.GetProperty(dbusInterface + ".state")
+		dbusTimeout = false
+	}(ctx)
+
+	wg.Wait()
+	if dbusTimeout {
```

Comment:
I'd have to have a closer look, but at a quick glance it feels a bit off to both have the `currentState` and the `dbusTimeout` . Wondering if we need both, or if it would work if we would conditionally set `currentState` (once `GetProperty` returns without an error).


Even wondering if the default value of `currentState` would be `nil`, and in that case we could possibly use that? (i.e., we were unable to get a value)

Reply:
_currentState_ does not always get set when the dbus API call hangs.  I used _dbusTimeout_ to make it clear when dbus timed out versus when _currentState_ was not set.  Also, I don't believe the default value of _currentState_ is nil.